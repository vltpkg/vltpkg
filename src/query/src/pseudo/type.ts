import { splitDepID } from '@vltpkg/dep-id/browser'
import type { ParserState } from '../types.ts'
import { asPostcssNodeWithChildren, asTagNode } from '../types.ts'
import { removeDanglingEdges, removeNode } from './helpers.ts'

/**
 * :type(str) Pseudo-Element will match only nodes that are of
 * the same type as the value used. The type is determined by the
 * first part of the dependency ID.
 */
export const type = async (state: ParserState) => {
  const type = asPostcssNodeWithChildren(state.current)
  const selector = asPostcssNodeWithChildren(type.nodes[0])
  const name = asTagNode(selector.nodes[0]).value
  for (const node of state.partial.nodes) {
    const nodeType = splitDepID(node.id)[0]
    if (nodeType !== name) {
      removeNode(state, node)
    }
  }
  removeDanglingEdges(state)
  return state
}
