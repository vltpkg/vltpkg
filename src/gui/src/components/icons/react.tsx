export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
