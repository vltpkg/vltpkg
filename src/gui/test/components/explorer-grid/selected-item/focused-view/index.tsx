import { test, expect, vi, beforeEach, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.ts'
import { FocusedView } from '@/components/explorer-grid/selected-item/focused-view/index.tsx'

import type { GridItemData } from '@/components/explorer-grid/types.ts'

vi.mock(
  '@/components/explorer-grid/selected-item/context.tsx',
  () => ({
    useSelectedItemStore: vi.fn(),
    SelectedItemProvider: 'gui-selected-item-provider',
  }),
)

vi.mock(
  '@/components/explorer-grid/selected-item/item-header.tsx',
  () => ({
    ItemHeader: 'gui-item-header',
  }),
)

vi.mock('@/components/explorer-grid/selected-item/item.tsx', () => ({
  SelectedItemTabs: 'gui-selected-item-tabs',
}))

vi.mock(
  '@/components/explorer-grid/selected-item/focused-view/aside.tsx',
  () => ({
    FocusedAside: 'gui-focused-aside',
  }),
)

vi.mock(
  '@/components/explorer-grid/selected-item/focused-view/focused-button.tsx',
  () => ({
    FocusButton: 'gui-focused-button',
  }),
)

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

beforeEach(() => {
  vi.clearAllMocks()
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

test('FocusedView renders default', () => {
  const mockItem = {
    id: '1',
    labels: ['prod'],
    name: 'item',
    title: 'item',
    version: '1.0.0',
    sameItems: false,
    stacked: false,
    size: 1,
  } satisfies GridItemData

  const mockDependencies: GridItemData[] = []
  const mockUninstalledDependencies: GridItemData[] = []
  const mockOnDependencyClick = vi.fn(() => () => undefined)

  const Container = () => {
    return (
      <FocusedView
        item={mockItem}
        dependencies={mockDependencies}
        onDependencyClick={mockOnDependencyClick}
        uninstalledDependencies={mockUninstalledDependencies}
      />
    )
  }

  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})
