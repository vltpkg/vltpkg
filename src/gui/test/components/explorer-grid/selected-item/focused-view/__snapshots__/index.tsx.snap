// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FocusedView renders default 1`] = `

<gui-selected-item-provider
  selecteditem="[object Object]"
  asideoveriewvisible="false"
>
  <section
    class="relative pb-4"
    style="opacity: 0; filter: blur(2px);"
  >
    <gui-item-header classnames="[object Object]">
      <gui-focused-button>
      </gui-focused-button>
    </gui-item-header>
    <div class="mx-6 grid grid-cols-12">
      <div class="col-span-full h-fit rounded-xl border-[1px] border-muted bg-white pt-2 dark:bg-neutral-900 lg:col-span-9">
        <gui-selected-item-tabs>
        </gui-selected-item-tabs>
      </div>
      <gui-focused-aside
        classname="mt-4 md:mt-0"
        dependencies
        uninstalleddependencies
      >
      </gui-focused-aside>
    </div>
  </section>
</gui-selected-item-provider>

`;
